CondCore/ESSources/test/SiStripTest/noiseanalyzer.cc

```diff
@@ -2,50 +2,49 @@
 #include <string>
 #include <iostream>
 #include <map>
-#include "FWCore/Framework/interface/EDAnalyzer.h"
-#include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/ESHandle.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
-#include "FWCore/Framework/interface/EventSetup.h"
 
 #include "CondFormats/Calibration/interface/mySiStripNoises.h"
 #include "CondFormats/DataRecord/interface/mySiStripNoisesRcd.h"
 
+#include "FWCore/Framework/interface/global/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
 using namespace std;
 
 namespace edmtest {
-  class NoisesAnalyzer : public edm::EDAnalyzer {
+  class NoisesAnalyzer : public edm::global::EDAnalyzer<> {
   public:
-    explicit NoisesAnalyzer(edm::ParameterSet const& p) { std::cout << "NoisesAnalyzer" << std::endl; }
-    explicit NoisesAnalyzer(int i) { std::cout << "NoisesAnalyzer " << i << std::endl; }
-    virtual ~NoisesAnalyzer() { std::cout << "~NoisesAnalyzer " << std::endl; }
-    virtual void analyze(const edm::Event& e, const edm::EventSetup& c);
+    explicit NoisesAnalyzer(edm::ParameterSet const& p) : theNoisesToken_(esConsumes()) {
+      edm::LogPrint("NoisesAnalyzer") << "NoisesAnalyzer";
+    }
+    explicit NoisesAnalyzer(int i) { edm::LogPrint("NoisesAnalyzer") << "NoisesAnalyzer " << i; }
+    virtual ~NoisesAnalyzer() { edm::LogPrint("NoisesAnalyzer") << "~NoisesAnalyzer "; }
+    void analyze(edm::StreamID, const edm::Event& e, const edm::EventSetup& c) const override;
 
   private:
+    const edm::ESGetToken<mySiStripNoises, mySiStripNoisesRcd> theNoisesToken_;
```

Comment:
@francescobrivio does this token ( and all the others added ) need to be initialise with esConsumes ?

Reply:
it's initialized few lines before (line 19):
```
explicit NoisesAnalyzer(edm::ParameterSet const& p) : theNoisesToken_(esConsumes())
```

Reply:
@francescobrivio thanks, did not noticed...